dvc/cache/base.py

```diff
@@ -101,17 +121,26 @@ def changed(self, path_info, hash_info):
         Args:
             path_info: dict with path information.
             hash: expected hash value for this data.
+            filter_info: an optional argument to target a specific path.
 
         Returns:
             bool: True if data has changed, False otherwise.
         """
 
+        if filter_info is not None and filter_info != path_info:
+            path = filter_info
+            path_info, hash_info = self._get_filtered_hash_info(
+                path, hash_info, path_info
+            )
+        else:
+            path = path_info
```

Comment:
Notice how we don't really need filtered hash_info until `changed_cache` call below. We shouldn't try to filter it until we really need it, as it might be costly for big datasets. The `trace()` message that uses it can be removed or modified to just include filter_info too, so you don't have to have filtered hash_info for it.